### ISSUE TITLE
Forgotten line of code blocking MAXIMIZED state

### ISSUE THREAD

**Issue by Pavel-N** [linebender/druid#2113]
In GTK backend in the file [window.rs in function `set_window_state()` there is a line of code](https://github.com/linebender/druid/blob/6d73a336eede48f95ecd9217533d02ca51a7467c/druid-shell/src/backend/gtk/window.rs#L1068)  that is succesfully preventing from going into maximized state.
```rust
pub fn set_window_state(&mut self, size_state: window::WindowState) {
        use window::WindowState::{MAXIMIZED, MINIMIZED, RESTORED};
        let cur_size_state = self.get_window_state();
        if let Some(state) = self.state.upgrade() {
            match (size_state, cur_size_state) {
                (s1, s2) if s1 == s2 => (),
                (MAXIMIZED, _) => state.window.maximize(),
                (MINIMIZED, _) => state.window.iconify(),
                (RESTORED, MAXIMIZED) => state.window.unmaximize(),
                (RESTORED, MINIMIZED) => state.window.deiconify(),
                (RESTORED, RESTORED) => (), // Unreachable
            }

            // state.window.unmaximize(); <--- When this line gets commented out it fixes everything
        }
    }
```
I removed it, ran `cargo clean` and re-compiled and everything now works as it should.
I guess it was just forgotten there?

**Comment by raphlinus** [comment]
It looks like it to me. @rjwittams is the original author, would you like to confirm. In any case, if you'd like to send a PR to fix it would be appreciated.

**Comment by Dequog** [comment]
Will you give me the steps to fix this in my project please?

**Comment by ssokolow** [comment]
@Dequog Download a copy of the repo, remove the line, and then add an appropriate [`[patch]` section](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section) to your `Cargo.toml`, I assume.

**Comment by Pavel-N** [comment]
@Dequog Also If you are using VSCode you can just `Ctrl+Left Click` on the `Window` struct to open the file (window.rs), there you can serach for the line using `Ctrl+f`, comment out / remove the line, save. Then you wanna run `cargo clean` and you are ready to go.

**Comment by Dequog** [comment]
@Pavel-N Thank you very much

**Comment by jneem** [comment]
Fixed by #2118
